src/plugins/intel_gpu/src/plugin/transformations/kv_cache_fusion.cpp

```diff
@@ -62,9 +63,9 @@ KVCacheFusionMatcher::KVCacheFusionMatcher() {
         if (past_node->get_variable_id() != present_node->get_variable_id())
             return false;
 
-        // TODO: Support conversion internally
-        if (!concat_node || concat_node->get_output_element_type(0) != past_node->get_output_element_type(0))
-            return false;
+        // // TODO: Support conversion internally
+        // if (!concat_node || concat_node->get_output_element_type(0) != past_node->get_output_element_type(0))
+        //    return false;
```

Comment:
It intents to make sure the inputs and output of concat are with the same data type. Instead of deleting the check (which would lead to potential risk to inference result and hard to debug), a better check here may be, I think, to check the data type consistency of concat_past_input->get_output_element_type(0) and concat_node->get_output_element_type(0).

Reply:
had fix, thank you very much.